### DISCUSSION TITLE
PCL Community 成员基本方针

### DISCUSSION THREAD

**Discussion by WForst-Breeze** [PCL-Community/.github#8] [公告 | Announcements]
👉 可以无需特别通知他人而进行的操作：

1. 在组织内创建仓库，并制作你喜欢的有关 Minecraft 或 PCL 的，甚至是其它的你喜欢的、有意义的东西；
2. 将你的仓库转移至组织名下；
3. 协助其他仓库，并为其贡献你的内容。注意，如果这个仓库不是你创建的，则请尽量使用 Pull Request 来贡献，以避免不必要的麻烦。

👉 需要特别通知组织 Owner 或相关方，获得同意后方可进行的操作：
1. 邀请新成员成为组织内 Member；
2. 邀请他人担任 Outside Collaborator；
3. 其他可能涉及到他人权益或劳动成果的操作。

👉 在任何情况下，任何成员绝对禁止的操作：

1. 对其他仓库进行破坏性的、恶意的改动；
2. 将无关仓库 Fork 在本组织名下；
3. 不经仓库创建者允许或未经仔细审查直接 Merge 已有的 Pull Request；
4. 创建意义不明的、占用社区资源的仓库；
5. 未经所在仓库的负责人允许，私自邀请不在组织内的成员担任 Outside collaborator；
6. 未经通知而更改组织成员的 Membership；
7. 其他破坏组织秩序和和谐的行为。

如若违反上述规定，视情节轻重，将会对其进行警告或直接踢出组织。

---

附：PCL Community 交流群管理条例

PCL Community 交流群欢迎任何 PCL 社区玩家加入，在加入后，你需要遵循以下规定：
1. 群名片设置为 GitHub ID
2. 禁止在开发相关话题开始或进行时歪楼闲聊
3. 不得进行任何形式的、对他人的人身攻击，包括但不限于：引战、辱骂、阴阳怪气等
4. 不得提及或讨论任何政治敏感或有关国家领导人的话题或字眼

如若违反上述规定，管理员有权在不进行任何提醒的情况下予以撤回消息、警告、禁言、踢出群聊、踢出组织等操作

**上述内容最终解释权均归相关负责人、管理员所有**

**Comment by MoYuan-CN** [comment]
👉 Pull Request 规范（暂行，适用于 PCL2 社区版各仓库）

1. 必须有至少 1 人进行了代码审查之后才可以合并 PR
2. 不涉及产品代码的 PR（例如改 Issue Template 等）请在最终提交信息处附加 `[skip ci]`
3. 涉及产品代码的 PR 请在提交标题内写规范 Commit Message，例如 `feat(ui): 新界面`；
提交正文内可以注明这是合并了哪一条 PR，往什么分支等

**Comment by Deep-Dark-Forest** [comment]
## 对组织内仓库直接 commit 适用条件的解释

**组织内任何恶意破坏性的 commit 是不被允许的。**
对组织内仓库直接 commit 适用条件取决于该仓库的类型。根据社区成员达成的共识、仓库负责人的定义，组织内仓库分为三类：
1. 组织公共仓库
例如 .github、.github-private 等，任何组织成员可以直接 commit。
2. 组织内多人协作仓库
例如 PCL2-CE，开发者、维护者不止一个的仓库，只有该仓库的负责人以及获得授权的人（如果仓库负责人未指定获权者，则按照社区成员的共识）在修改任何数量的内容时可以直接 commit，其他人只有在更改少量不影响代码本身的内容（如 README.md，代码注释等）可以直接 commit，但仓库负责人另有说明的除外。
3. 组织内单人维护仓库
完全由仓库负责人维护、开发的仓库，只有该仓库的负责人在修改任何数量的内容时可以直接 commit，其他人只有在更改少量且不影响代码本身的内容（如 README.md，代码注释等）可以直接 commit，但仓库负责人另有说明的除外。
4. 本解释中所有“组织内仓库”均包括仓库内的所有分支、所有文件。

**Comment by Deep-Dark-Forest** [comment]
组织内禁止创建违反中国大陆地区法律法规的内容。一经发现社区管理员有权立即删除内容并将有关人员移出组织。涉及行政或刑事违法的，我们会尽力配合公安机关等政府部门的调查。本组织不对成员创建的违法项目承担任何法律责任。
